Userland/Libraries/LibELF/DynamicLoader.cpp

```diff
@@ -527,6 +534,23 @@ DynamicLoader::RelocationResult DynamicLoader::do_direct_relocation(DynamicObjec
         return VirtualAddress { reinterpret_cast<DynamicObject::IfuncResolver>(address.get())() };
     };
 
+    struct ResolvedTLSSymbol {
+        FlatPtr value;
+        DynamicObject const* dynamic_object;
+    };
+
+    auto resolve_tls_symbol = [](DynamicObject::Relocation const& relocation) -> Optional<ResolvedTLSSymbol> {
+        if (relocation.symbol_index() != 0) {
```

Comment:
It didn't matter too much where the code was previously, but maybe inverting this (i.e. some kind of early return) makes more sense when factoring it out into a separate function.

Reply:
Good catch, it's nicer that way. Thanks!